lake/oneshot_cls.py

```diff
@@ -263,7 +263,12 @@ def main():
     recall_target = torch.from_numpy(np.array(recall_target)).to(device)
 
     # Reset to saved model
-    model.load_state_dict(torch.load(pretrained_model_path))
+    model_dict = model.state_dict()
+    pretrained_dict = torch.load(pretrained_model_path)
+    pretrained_dict = {k: v for k, v in pretrained_dict.items() if k in model_dict}
```

Comment:
why is this necessary?

Reply:
Optional to keep it, but it was motivated by the fact that I wanted to use a pretrained LTM (that had the PM in it) but PyTorch refuses to load it because I hadn't built the PM.

This line of code essentially only loads the parts that you actually initalised by checking against the instantiated model and comparing it with what's being loaded from the pretrained path.